tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -270,6 +270,25 @@ public TracerSettings(IConfigurationSource source)
             {
                 HttpClientExcludedUrlSubstrings = TrimSplitString(urlSubstringSkips.ToUpperInvariant(), commaSeparator);
             }
+
+            DbmPropagationLevel dbmPropagationValue;
+            DbmPropagationMode = Enum.TryParse(source?.GetString(ConfigurationKeys.DbmPropagationMode), true, out dbmPropagationValue) ?
+                $"{dbmPropagationValue}".ToLower() : "disabled";
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the tracer should propagate service data in db queries
+        /// </summary>
+        public enum DbmPropagationLevel
```

Comment:
Be really careful about what could end up in the public api
```suggestion
        internal enum DbmPropagationLevel
```